qiskit/transpiler/target.py

```diff
@@ -484,6 +484,28 @@ def operations_for_qargs(self, qargs):
             raise KeyError(f"{qargs} not in target.")
         return [self._gate_name_map[x] for x in self._qarg_gate_map[qargs]]
 
+    def instruction_supported(self, operation_name, qargs):
+        """Return whether the instruction (operation + qubits) is supported by the target
+
+        Args:
+            operation_name (str): The name of the operation for the instruction
+            qargs (tuple): The tuple of qubit indices for the instruction
+
+        Returns:
+            bool: Returns ``True`` if the instruction is supported and ``False`` if it isn't.
+
+        """
+        # Case a list if passed in by mistake
+        qargs = tuple(qargs)
+        if operation_name in self._gate_map:
+            if qargs in self._gate_map[operation_name]:
+                return True
+            if self._gate_map[operation_name] is None or None in self._gate_map[operation_name]:
+                return self._gate_name_map[operation_name].num_qubits == len(qargs) and all(
```

Comment:
On second thoughts, perhaps checking for `Instruction.num_qubits` is not ideal if we anticipate needing to handle any sort gate with variadic arguments, e.g. `mcrx` or so.  `Instruction` lets you do that by having different `num_qubits` for the different instances, so the instance attribute isn't necessarily applicable to all objects that share the same gate name.

We don't really have a formal statement of our assumptions about the properties of `Instruction`, though, and I think other parts of Terra probably assume that `name: num_qubits` is a many-to-one mapping as well, so I'm ok either way.

Reply:
I think as far as the target goes this is fine. If an mcrx gate is in the target it will have a fixed number of qubits as it's set at instantiation, and the target only stores an instance of the operation object as that not the class is what represents the implementation on the backend. Even if this wasn't the case though and it was a class level thing the target and transpiler don't support transpiling to variable width operations yet (I'm thinking of something like global Mølmer–Sørensen gate which an ion system could implement as a native instruction and we'd need a fixed named variant for each number of qubits for the transpiler to work with it). This is something we discussed when creating the target and it's a known limitation with the current implementation, but the target was was a huge incremental step forward over the previous model and provides a great deal more flexibility and expressiveness than was previous available so even if we can't fully model every use case having just everything with fixed sizes for now is fine. Once we get the transpiler up to snuff and fully leveraging what we can express today I think we can look at adding variable width gates to the target model and the transpiler.

Reply:
Yep, that all sounds plenty convincing to me.  I'll merge this now so you can use the new method like you wanted in other PRs.